### PR TITLE
Use deserializer to parse size in human readable format from fileconfig

### PR DESCRIPTION
**Pull request by sum12**
This patch allows the disk cache size to be specified in the similar
format as SCCACHE_CACHE_SIZE. Human readable format with suffix K/M/G/T.

To maintain backwards compatibility a visit_u64 is implemented as a
transparent fn.  The patch also addresses the requested TODO

Added tests for the corresponding part

### CODE REVIEW

**Comment by sum12** [comment]
The Failure seems unreleated to the change.

**Comment by Xanewok** [comment]
It seems like CI base images for dist tests need to be updated, see https://github.com/paritytech/sccache/pull/74

File: src/config.rs [hunk 1]
```diff
@@ -145,12 +146,43 @@ impl HTTPUrl {
...
+where
+    D: de::Deserializer<'de>,
+{
+    struct Visitor;
+
+    impl<'de> de::Visitor<'de> for Visitor {
+        type Value = u64;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+            formatter.write_str("string disk size with suffix K/M/G/T or integer for Bytes ")
```
Comment (mitchhentges) [inline src/config.rs:159]: I'm not sufficiently familiar with `serde`, so just to confirm: is the extra space at the end of this string deliberate?

**Review by mitchhentges — APPROVED** [review]
Looks good, thanks!
Can you rebase this against `main` so that we can have modern CI run against this change?

File: src/config.rs (hunk 1 above)
Comment (sum12) [reply src/config.rs]: thanks, fixed it

**Comment by mitchhentges** [comment]
Heads up that there's some CI failures that need to be resolved here first before I can land this.

**Comment by drahnr** [comment]
Gentle ping.

**Review by drahnr — APPROVED** [review]

**Comment by sylvestre** [comment]
sorry, I make a mistake, could you please resend/recreate it if you still want to land it?

**Pull request closed without merging**